### ISSUE TITLE
HiKey, pager: panic at core/mm/vm.c:735 <vm_set_prot>

### ISSUE THREAD

**Issue by jforissier** [OP-TEE/optee_os#4624]
Reproducing an issue seen a few times during IBART runs (such as [here](https://optee.mooo.com:5000/logs/OP-TEE/optee_os/4606/641462945/07508bff9a1cc9f19a47e380999fc8fcfaf5ee58#end-01-hikey6220-test) or [here](https://optee.mooo.com:5000/logs/OP-TEE/optee_os/4608/642900949/7c5f41ae14c4dc43cbceaf6bcedfcecb1d876383#end-01-hikey6220-test)):
```
$ make -j10 CFG_RPMB_WRITE_KEY=y CFG_RPMB_FS=y CFG_SECURE_DATA_PATH=y CFG_WITH_PAGER=y CFG_TEE_CORE_LOG_LEVEL=2 CFG_CORE_MBEDTLS_MPI=y CFG_ENCRYPT_TA=y GP_PACKAGE=~/work/TEE_Initial_Configuration-Test_Suite_v2_0_0_2-2017_06_09.7z CFG_ENABLE_EMBEDDED_TESTS=y CFG_IN_TREE_EARLY_TAS=pkcs11/fd02c9da-306c-48c7-a49c-bbd827ae86ee CFG_PKCS11_TA=y
$ make flash
```
```
$ while [ 1 ]; do xtest regression_1010; done
```
```
  regression_1010.8 OK
o regression_1010.9 Invalid memory access 4 with 1024 bytes memref
E/TC:7 0 Panic at core/mm/vm.c:735 <vm_set_prot>
E/TC:7 0 TEE load address @ 0x3f000000
E/TC:7 0 Call stack:
E/TC:7 0  0x000000003f0062d0 print_kernel_stack at optee_os/core/arch/arm/kernel/unwind_arm64.c:80
E/TC:7 0  0x000000003f00c9d8 __do_panic at optee_os/core/kernel/panic.c:24
E/TC:7 0  0x000000003f07cfac vm_set_prot at optee_os/core/mm/vm.c:735
E/TC:7 0  0x000000003f076a64 ldelf_syscall_map_bin at optee_os/core/kernel/ldelf_syscalls.c:380
E/TC:7 0  0x000000003f071928 tee_svc_do_call at optee_os/core/arch/arm/tee/arch_svc_a64.S:140
E/TC:7 0  0x000000003f005148 thread_svc_handler at optee_os/core/arch/arm/kernel/thread.c:1586 (discriminator 4)
E/TC:7 0  0x000000003f003940 el0_svc at optee_os/core/arch/arm/kernel/thread_a64.S:653
```
The crash is not always in the same test, but the call stack seems to be always identical.

**Comment by jforissier** [comment]
As mentioned by @etienne-lms in https://github.com/OP-TEE/optee_os/pull/4608#issuecomment-842937437, commit 4a3f6ad054d4 ("core: pager: let struct tee_pager_area span multiple translation tables") is indeed the first bad one.

**Comment by jenswi-linaro** [comment]
The problem is 100% reproducible on QEMU v7 LPAE=n and pager enabled. With this patch in optee_test
``` diff
--- a/ta/os_test/os_test.c
+++ b/ta/os_test/os_test.c
@@ -101,6 +101,8 @@ static TEE_Result get_binblock_property(TEE_PropSetHandle h,
        return res;
 }
 
+const uint8_t hej[1 * 1024 * 1024] = { 1 };
+
 static TEE_Result print_properties(TEE_PropSetHandle h,
                                   TEE_PropSetHandle prop_set,
                                   struct p_attr *p_attrs, size_t num_p_attrs)
@@ -122,6 +124,8 @@ while (true) {
        char *bbuf = NULL;
        size_t bblen = 0;
 
+       DMSG("%p %zu", (void *)hej, sizeof(hej));
+
        res = TEE_GetPropertyName(h, nbuf, &nblen);
        if (res != TEE_SUCCESS) {
                EMSG("TEE_GetPropertyName returned 0x%x\n",
```

xtest 1010.

**Comment by jforissier** [comment]
Fixed in #4628.
